res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,718 @@
+    if (value === DOWN) {
+        if (engine.getValue(group, "track_samples")) { //Sampler loaded (playing or not)
+            engine.setValue(group, "cue_gotoandplay", 1);
+        } else {
+            engine.setValue(group, "LoadSelectedTrack", 1);
+        }
+    }
+};
+
+ReloopBeatmix24.ShiftSamplerPad = function(channel, control, value, status, group) {
+    if (value === DOWN) {
+        if (engine.getValue(group, "track_samples")) { //Sampler loaded (playing or not)
+            if (engine.getValue(group, "play")) { // Sampler is playing
+                engine.setValue(group, "cue_gotoandstop", 1);
+            } else {
+                engine.setValue(group, "eject", 1);
+            }
+        } else {
+            engine.setValue(group, "LoadSelectedTrack", 1);
+        }
+    }
+};
+
+ReloopBeatmix24.SamplerVol = function(channel, control, value, status, group) {
+    for (var i = 1; i <= engine.getValue("[Master]", "num_samplers"); i++) {
+        engine.setValue("[Sampler" + i + "]", "volume", value / 127.0);
+    }
+};
+
+// Jog Wheel functions
+// ========================================================
+
+ReloopBeatmix24.WheelTouch = function(channel, control, value, status, group) {
+    var deck = parseInt(group.substr(8, 1), 10);
+    if (value === DOWN) {
+        var alpha = 1.0 / 8;
+        var beta = alpha / 32;
+        engine.scratchEnable(deck, 1600, 33 + 1 / 3, alpha, beta);
+    } else {
+        engine.scratchDisable(deck);
+    }
+};
+
+ReloopBeatmix24.WheelTurn = function(channel, control, value, status, group) {
+    var newValue = value - 64;
+    var deck = parseInt(group.substr(8, 1), 10);
+
+    // In either case, register the movement
+    if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, newValue); // Scratch!
+    } else {
+        engine.setValue(group, 'jog', newValue); // Pitch bend
+    }
+};
+
+// Led Feedback functions
+// ========================================================
+ReloopBeatmix24.AllJogLEDsToggle = function(deck, state, step) {
+    var step = typeof step !== 'undefined' ? step : 1; // default value
+    for (var j = 0x30; j <= 0x3F; j += step) {
+        midi.sendShortMsg(deck, j, state);
+    }
+};
+
+ReloopBeatmix24.deckloaded = function(value, group, control) {
+    var i;
+    switch (group.substr(1, 7)) {
+        case "Channel":
+            var channelRegEx = /\[Channel(\d+)\]/;
+            var channelChan = parseInt(channelRegEx.exec(group)[1]);
+            if (channelChan <= 4) {
+                midi.sendShortMsg(0x90 + channelChan, 0x50, value ? ON :
+                    OFF);
+            }
+            break;
+        case "Sampler":
+            var samplerRegEx = /\[Sampler(\d+)\]/;
+            var samplerChan = parseInt(samplerRegEx.exec(group)[1]);
+            if (samplerChan <= 8) { // We only handle 8 samplers (1 per pad)
+                for (i = 0x91; i <= 0x94; i++) {
```

Comment:
Because in sampler mode, all pads (the 4 channel pads) control the same samplers. I have chosen to control 8 samplers, not 8x4 samplers.
so when a sampler is loaded, I need to lits 8 leds. 2 per channel (shifted and unshifted)